tests/kafkatest/services/templates/kafka.properties

```diff
@@ -20,6 +20,8 @@ port=9092
 #host.name=localhost
 advertised.host.name={{ node.account.hostname }}
 #advertised.port=<port accessible by clients>
+listeners=PLAINTEXT://:9092,SSL://:9093
```

Comment:
I think we're going to have to protect all of this SSL stuff in conditionals (both blocks of settings added in this patch). Otherwise the version upgrade tests in KAFKA-1888 will break since these settings are not backwards compatible.

We can of course address this in KAFKA-1888 (which has an existing patch already), but it seems more relevant to add in this patch since this is the one adding non-default settings.

Also, do we want to have cases where we *only* have SSL? It seems like cases where we're explicitly testing SSL should only turn on SSL so we don't, e.g., accidentally pass by using `PLAINTEXT` by mistake.